omc3/optics_measurements/beta_from_phase.py

```diff
@@ -419,21 +419,33 @@ def three_bpm_method(meas_input, phase, plane, meas_and_mdl_tunes):
     bet_frac = (cot_phase_meas_shift1[0]/cot_phase_model_shift1[0] +
                 cot_phase_meas_shift1[3]/cot_phase_model_shift1[3] +
                 cot_phase_meas_shift2[1]/cot_phase_model_shift2[1]) / 3
+
+    alf_mdl_term = (((cot_phase_model + np.roll(cot_phase_model, -1, axis=0))[0] + 2.0 * alfmdl)
+                + ((cot_phase_model + np.roll(cot_phase_model, -1, axis=0))[3] + 2.0 * alfmdl)
+                + ((cot_phase_model + np.roll(cot_phase_model, -2, axis=0))[1] + 2.0 * alfmdl)) / 6.0
+    alf_meas_term = (((cot_phase_meas + np.roll(cot_phase_meas, -1, axis=0))[0])
+                + ((cot_phase_meas + np.roll(cot_phase_meas, -1, axis=0))[3])
+                + ((cot_phase_meas + np.roll(cot_phase_meas, -2, axis=0))[1])) / 6.0
+
     # multiply the fractions by betmdl and calculate the arithmetic mean
     beti = bet_frac * betmdl
-    alfi = (bet_frac * (cot_phase_model[1] + cot_phase_model[3] + 2 * alfmdl) - (cot_phase_meas[1] + cot_phase_meas[3])) / 2
+    alfi = bet_frac * alf_mdl_term + alf_meas_term
```

Comment:
Should have just the average, or some weighting, for example, based on phase advances? This could avoid divergent behaviour.